cirq-core/cirq/circuits/moment.py

```diff
@@ -659,37 +659,26 @@ def cleanup_key(key: Any) -> Any:
         return diagram.render()
 
     def _commutes_(self, other: Any, *, atol: float = 1e-8) -> Union[bool, NotImplementedType]:
-        """Determines whether Moment commutes with the Operation.
+        """Determines whether Moment commutes with the other Moment or Operation.
 
         Args:
-            other: An Operation object. Other types are not implemented yet.
-                In case a different type is specified, NotImplemented is
-                returned.
+            other: An Operation or Moment object to test for commutativity.
             atol: Absolute error tolerance. If all entries in v1@v2 - v2@v1
                 have a magnitude less than this tolerance, v1 and v2 can be
                 reported as commuting. Defaults to 1e-8.
 
         Returns:
-            True: The Moment and Operation commute OR they don't have shared
-            quibits.
+            True: The Moment commutes with Moment or Operation OR they don't
+                have shared qubits.
             False: The two values do not commute.
             NotImplemented: In case we don't know how to check this, e.g.
-                the parameter type is not supported yet.
+                the parameter type is not supported or commutativity cannot be
+                determined.
         """
-        if not isinstance(other, ops.Operation):
+        if not isinstance(other, (ops.Operation, Moment)):
             return NotImplemented
-
-        other_qubits = set(other.qubits)
-        for op in self.operations:
-            if not other_qubits.intersection(set(op.qubits)):
-                continue
-
-            commutes = protocols.commutes(op, other, atol=atol, default=NotImplemented)
-
-            if not commutes or commutes is NotImplemented:
-                return commutes
-
-        return True
+        other_operations = other.operations if isinstance(other, Moment) else (other,)
```

Comment:
Is there a generic way to get a seq of ops from a thing? It'd be nice to be able to check commutativity between various things that can hold operations, rather than limit to moments or single ops here.

Reply:
Ack.  The `_operations_commutes_impl` assumes that the compared lists of operations have no double-acted qubits which is guaranteed for Moment.operations.  We already fix and extend the previous API here.  I'd rather leave commutation with other types for later when we run into such needs.

Reply:
It occurred to me that maybe `OP_TREE` would be a better input type for the new function, since it supports single ops or op lists. Though it might be a teensy bit slower. Anyway, minor point.